### DISCUSSION TITLE
Funkwale integration

### DISCUSSION THREAD

**Discussion by luckyzor** [music-assistant/support#1750] [Feature requests and ideas]
Hello all, 
Really looking forward for this project and I wanted to integrate my funkwale library on this.
Is there any possibility to have this in the future?

https://www.funkwhale.audio/

Funkwhale is a self-hosted audio player and publication platform. It enables users to build libraries of existing content and publish their own.

Thank you all for everything 😊

**Comment by OzGav** [comment]
Someone would have to pickup this idea. Additionally I had not heard of this before but seems there has been some discussion on the web about copyright issues. We wouldn’t get into anything that is questionable.

**Reply by maste9** [reply]
Hi @OzGav,
There's not more copyright infringement risk than in any local music repository I would say.
I have a funkwhale server running on my homeserver - which is btw not federating - which allows me to stream my music from anywhere I have internet. My libraries there are private so that only me and my partner can listen to it.
So a provider that allows me to login to those accounts by setting domain, username, password would in deed be nice. But I don't know how easy the api would be to integrate into MA actually ...

@luckyzor As my MA and my funkwhale server run on the same server I worked around that by adding the media directory that's used by fw as a read-only local directory in MA. But that's of course a special situation with a self-hosted fw.

**Reply by luckyzor** [reply]
Unfortunately my funkwale server is not on the same server, so I really need the integration.
Concerning copyright issues, funkwale is the same as plex, local files and other self hosted solutions.
Hope someone with skills will add this to music assistant if it's possible.

**Reply by OzGav** [reply]
Yeah ok I can understand that you are using this for your own private purposes. Obviously it has functionality beyond that though.

Having said that being able to stream to the HA companion apps is on the todo list. This would allow you to achieve what you can do now I believe, that is, listen to your music wherever you are. Not sure when that functionality will be available though.

**Reply by khers** [reply]
I know that funkwhale supports a limited subsonic api. We have a subsonic provider working and except for got top songs by artist and get similar songs funkwhale supports everything it needs. We can change the subsonic provider to make the top songs and similar songs optional and everything should "just work" ;)

Please report anything that doesn't and it'll get fixed.

**Comment by OzGav** [comment]
I am wondering if this is now a redundant request as Remote Access has been added in MA 2.7 and above?
